lsp-verilog.el

```diff
@@ -0,0 +1,78 @@
+;;; lsp-verilog.el --- Verilog Client settings         -*- lexical-binding: t; -*-
+
+;; Copyright (C) 2019 Patrick Grogan
+
+;; Author: Patrick Grogan <pogrogan@gmail.com>
+;; Created: 7 December 2019
+;; Keywords: languages, lsp, verilog
+
+;; This program is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+;;; Commentary:
+;; LSP client support for Verilog. Right now, the only supported LSP server
+;; is HDL CHecker. See https://github.com/suoto/hdl_checker
+;;
+;; This file is based on the lsp-vhdl.el file.
+;;
+;; Set the lsp-verilog-server-path to the binary directory if it is not
+;; in the User path;
+
+;;; Code:
+
+(require 'lsp-mode)
+
+(defvar hdl-checker-bin-name "hdl_checker"
+  "Name of HDL Checker binary.")
+
+(defvar hdl-checker-disp-name "lsp-hdl-checker"
+  "Display name for HDL Checker.")
+
+(defgroup lsp-verilog nil
+  "LSP support for Verilog/SystemVerilog."
+  :group 'lsp-mode)
+
+(defcustom lsp-verilog-server-path nil
+  "Path to binary server file."
+  :group 'lsp-verilog
+  :risky t
+  :type 'file)
+
+(defvar lsp-verilog--params nil)
+
+(defun lsp-verilog--create-connection ()
+  "Returns lsp-stdio-connection based on the selected server"
+  (lsp-verilog--set-server-path)
+  (lsp-verilog--set-server-args)
+  (plist-put
+   (lsp-stdio-connection (lambda () (list (plist-get lsp-verilog--params 'server-path) (plist-get lsp-verilog--params 'server-args))))
+   :test? (lambda () (f-executable? (plist-get lsp-verilog--params 'server-path)))))
```

Comment:
I think that :test? wont be needed here since the default implementation will do the same ( it takes the first param and uses executable-find to check whether it is ok).